### ISSUE TITLE
Immediate calls of zero-arg functions are parsed incorrectly

### ISSUE THREAD

**Issue by jwhitfieldseed** [jwhitfieldseed/peach#6]
```
❯ () => 2
λ: anonymous: () -> Number
❯ (() => 2)
λ: anonymous: () -> Number
❯ ((() => 2))
2: Number
```

Line 2 `(() => 2))` should call the function, resulting in type `Number`.

**Comment by jwhitfieldseed** [comment]
This will be fixed with the last part of #14, to remove significant parentheses from the function call syntax.

**Comment by jwhitfieldseed** [comment]
This will be fixed with #23
